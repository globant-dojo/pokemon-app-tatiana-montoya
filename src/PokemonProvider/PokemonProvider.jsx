/* External */
import React, { useEffect } from "react";
import { useState } from "react";

const PokemonContext = React.createContext();

const PokemonProvider = (props) => {
  const [pokemonMessages, setPokemonMessages] = useState({});
  useEffect(() => {
    fetch("/languages/es.json")
      .then((res) => res.json())
      .then((data) => {
        setPokemonMessages(data);
      });
  }, []);

  return (
    <PokemonContext.Provider value={{ POKEMON_MESSAGES: pokemonMessages }}>
      {props.children}
    </PokemonContext.Provider>
  );
};

export { PokemonProvider, PokemonContext };
