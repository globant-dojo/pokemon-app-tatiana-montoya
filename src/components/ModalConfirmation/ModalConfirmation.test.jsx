/* External */
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

/* Componentes */
import { ModalConfirmation } from ".";

/* Other */
import { PokemonContext } from "@/PokemonProvider";

describe("Tests on <ModalConfirmation />", () => {
  const message = "Prueba";
  const setOpenModal = jest.fn();
  const handleConfirmed = jest.fn();

  const renderModalConfirmation = () =>
    render(
      <PokemonContext.Provider value={{ POKEMON_MESSAGES: {} }}>
        <ModalConfirmation
          message={message}
          setOpenModal={setOpenModal}
          handleConfirmed={handleConfirmed}
        />
      </PokemonContext.Provider>
    );

  test("it should show the message", () => {
    renderModalConfirmation();

    expect(screen.findByText(message)).toBeTruthy();
  });

  test("it should call the setOpenModal when clicked on cancel", () => {
    renderModalConfirmation();

    const button = screen.getByRole("button", { name: "btnCancel" });

    fireEvent.click(button);

    expect(setOpenModal).toHaveBeenCalled();
    expect(setOpenModal).toHaveBeenCalledTimes(1);
  });

  test("it should call the handleConfirmed when clicked on confirm", () => {
    renderModalConfirmation();

    const button = screen.getByRole("button", { name: "btnConfirm" });

    fireEvent.click(button);

    expect(handleConfirmed).toHaveBeenCalled();
    expect(handleConfirmed).toHaveBeenCalledTimes(1);
  });

  test("it should match with the snapshot", () => {
    const { container } = renderModalConfirmation();

    expect(container).toMatchSnapshot();
  });
});
