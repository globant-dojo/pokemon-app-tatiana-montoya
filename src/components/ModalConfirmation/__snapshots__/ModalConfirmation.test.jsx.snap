// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <ModalConfirmation /> it should match with the snapshot 1`] = `
<div>
  <form
    class="pkm-modal-confirmation"
  >
    <p
      class="pkm-modal-confirmation-p"
    >
      Prueba
    </p>
    <button
      aria-label="btnConfirm"
      class="pkm-btn"
      type="button"
    />
    <button
      aria-label="btnCancel"
      class="pkm-btn"
      type="button"
    />
  </form>
</div>
`;
